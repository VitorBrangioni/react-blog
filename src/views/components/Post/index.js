import React from "react";
import { useHistory } from "react-router-dom";
import DateHelper from "../../../helpers/DateHelper";
import {
  Container,
  Box,
  ContainerTitle,
  ContainerBody,
  Button,
  Title,
  Body,
  Legend,
} from "./styles";

const Post = ({ post, testId }) => {
  const history = useHistory();
  const { title, body, author, metadata } = post;
  const { publishedAt } = metadata;

  return (
    <Container data-testid={testId}>
      <Box>
        <ContainerTitle>
          <Title>{title}</Title>
        </ContainerTitle>
        <ContainerBody>
          <Body>{`${body.slice(0, 200)}...`}</Body>
          <Legend>
            <strong>Author:</strong> {author.name} | <strong>Date: </strong>
            {DateHelper.formatDatetime(publishedAt)}
          </Legend>
        </ContainerBody>

        <Button
          onClick={() => history.push("/post", { post })}
          data-testid="btn-open-post"
        >
          Ler mais
        </Button>
      </Box>
    </Container>
  );
};

export default Post;
