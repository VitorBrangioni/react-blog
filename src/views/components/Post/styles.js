import styled from "styled-components";

export const Container = styled.div`
  margin: auto;
  width: 100%;
  padding: 20px;
  height: 500px;
`;

export const Box = styled.div`
  position: relative;
  background: linear-gradient(to bottom right, #871ae5 40%, #ff5824 100%);
  border-radius: 20px;
  box-shadow: 5px 5px 20px rgba(#cbcdd3, 10%);
  padding: 20px;
  height: 100%;
`;

export const ContainerTitle = styled.div`
  position: relative;
  width: 100%;
  height: 25%;
  text-align: center;
  color: white;
`;

export const ContainerBody = styled.div`
  padding: 0px 10px 0px 20px;
  position: relative;
  width: 100%;
  heigth: 100px;
  text-align: center;
  color: white;
  font-size: 1.6rem;
`;

export const ContainerButton = styled.div`
  position: relative;
  padding: 20px;
  heigth: 50px;
`;

export const Button = styled.button`
  position: absolute;
  background: #212121;
  width: 50%;
  height: 12%;
  border-radius: 20px;
  top: 80%;
  left: 25%;
  outline: 0;
  border: none;
  box-shadow: 2px 2px 10px rgba(#777777, 0.5);
  transition: all 0.5s ease-in-out;
  font-size: 1.4rem;
  color: white;
  cursor: pointer;
  &:hover {
    transform: scale(1.05);
    transition: all 0.3s ease-in-out;
  }
`;

export const Title = styled.h2``;

export const Body = styled.p`
  font-size: 1.8rem;
`;

export const Legend = styled.p`
  color: #212121;
  font-size: 1.6rem;
`;
