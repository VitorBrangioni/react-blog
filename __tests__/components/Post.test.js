import React from "react";
import Post from "../../src/views/components/Post";
import postsMock from "../../__mocks__/posts.mock";
import { shallow } from "enzyme";
import { render, fireEvent } from "@testing-library/react";
const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("Post", () => {
  it("PostsListLoader snapshot", () => {
    const wrapper = shallow(<Post post={postsMock[0]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Verify if click in button, change route to `post`", async () => {
    const { getByTestId } = await render(<Post post={postsMock[0]} />);
    const rightClick = { button: 0 };
    fireEvent.click(getByTestId("btn-open-post"), rightClick);

    expect(mockHistoryPush).toHaveBeenCalledWith("/post", {
      post: postsMock[0],
    });
  });
});
