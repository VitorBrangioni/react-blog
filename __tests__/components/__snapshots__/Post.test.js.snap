// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loaders PostsListLoader snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.h2>
        AssCo has revamped the theory of versioning
      </styled.h2>
    </styled.div>
    <styled.div>
      <styled.p>
        We will mesh the buzzword "strategic". Our feature set is unparalleled, but our subscriber-defined CAE and non-complex configuration is invariably considered a remarkable achievement. What does the co...
      </styled.p>
      <styled.p>
        <strong>
          Author:
        </strong>
         
        Zinnia Rickard
         | 
        <strong>
          Date: 
        </strong>
        2017-04-12
      </styled.p>
    </styled.div>
    <styled.button
      onClick={[Function]}
    >
      Ler mais
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`Post PostsListLoader snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.h2>
        AssCo has revamped the theory of versioning
      </styled.h2>
    </styled.div>
    <styled.div>
      <styled.p>
        We will mesh the buzzword "strategic". Our feature set is unparalleled, but our subscriber-defined CAE and non-complex configuration is invariably considered a remarkable achievement. What does the co...
      </styled.p>
      <styled.p>
        <strong>
          Author:
        </strong>
         
        Zinnia Rickard
         | 
        <strong>
          Date: 
        </strong>
        2017-04-12
      </styled.p>
    </styled.div>
    <styled.button
      data-testid="btn-open-post"
      onClick={[Function]}
    >
      Ler mais
    </styled.button>
  </styled.div>
</styled.div>
`;
